src/scheduler/worker.rs

```diff
@@ -74,6 +95,10 @@ impl<C: Context> Worker<C> {
         self.local().init(tls);
         self.parked.store(false, Ordering::SeqCst);
         loop {
+            while let Some((bucket, mut work)) = self.local_work_buffer.pop() {
+                debug_assert!(self.scheduler.work_buckets[bucket].is_activated());
```

Comment:
Do we allow adding work to the local buffer for buckets that are not yet activated? If we allow that, we have no way to guarantee that the buckets are activated when we pop the work, and the assertion here will fail.

Reply:
Ideally we allow pushing to local buckets under all circumstances, and block consuming local buckets if they're not globally open.

Perhaps one way to do that:   have two lists of references to local buckets, one for consuming, one for producing.   The producing ones are always available (since it always safe to produce locally).   The consuming ones are empty and only get populated when the respective bucket is noticed as being open according to the existing protocols.   When it is open, consuming and producing refer to the same local bucket.

Reply:
Now only activated work packets can be cached locally.

@qinsoon I think this assertion holds now, given the fix above.

Since:
1. A work packet has to be activated before caching it locally.
2. Locally-cached work packets are executed with the highest priority.
3. A worker will never yield unless it drains all its local work packets.
4. A stop-the-world bucket will never close until the end of the GC.
5. A bucket with open only if (bucket of the previous stage is empty && all workers are yielded)

So ideally,
1. Based on [1,4], when a local-work is being executed, its bucket is already open and will never close (until the end of GC)
2. Based on [2,3,5], a worker can never yield before it drains all the local works. Thus no more buckets will open.